### PR TITLE
Adds config deprecation system

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl:
server: Logs and admin messages will now appear when you are using deprecated config settings and advise on upgrades
/:cl:

A lot of our flags are inverted from sane defaults. I plan on fixing this in the future, this helps with that and any other future config changes we need to make.

### CODE REVIEW

**Comment by MrStonedOne** [comment]
>A lot of our flags are inverted from sane defaults. I plan on fixing this in the future, this helps with that and any other future config changes we need to make.

There are two kinds of defaults in configs

What do you mean by this?

**Comment by Cyberboss** [comment]
All flag type entries default to off but then we have stuff like ENABLE_AI
and ADMIN_LEGACY_SYSTEM and many more which don't make sense to have off by
default.

**Comment by MrStonedOne** [comment]
Again, what do you mean by default? There is default in the config (as in, what is in the repos config), and the default in code for when a config item doesn't exist in the config.

I call these new server defaults and existing server defaults

Please specify which default you are referring to when talking about configs

**Comment by Cyberboss** [comment]
The in code defaults, for when there are no config files.

**Comment by optimumtact** [comment]
@MrStonedOne happy with this?

File: code/controllers/configuration/configuration.dm [hunk 1]
```diff
@@ -123,11 +124,26 @@
...
+				warning("[new_ver.type] is deprecated but gave no proper return for DeprecationUpdate()")
+			
 		var/validated = E.ValidateAndSet(value)
 		if(!validated)
 			log_config("Failed to validate setting \"[value]\" for [entry]")
-		else if(E.modified && !E.dupes_allowed)
-			log_config("Duplicate setting for [entry] ([value], [E.resident_file]) detected! Using latest.")
+		else 
+			if(E.modified && !E.dupes_allowed)
+				log_config("Duplicate setting for [entry] ([value], [E.resident_file]) detected! Using latest.")
```
Comment (MrStonedOne) [inline code/controllers/configuration/configuration.dm]: As an unrelated aside this needs to go.

All configs should be designed such that they can be overridden to allow for the synced configs model /tg/ is using.

**Review by MrsTonedOne — APPROVED** [review]

**Pull request merged**